homeassistant/components/xiaomi_miio/sensor.py

```diff
@@ -135,17 +155,234 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
 
-        device = AirQualityMonitor(host, token)
-        entities.append(XiaomiAirQualityMonitor(name, device, config_entry, unique_id))
+        if model == MODEL_AIRPURIFIER_3C:
+            air_purifier = AirPurifierMB4(host, token)
+            entities.append(
+                XiaomiAirPurifier(
+                    NAME_AIR_QUALITY_INDEX, air_purifier, config_entry, unique_id
+                )
+            )
+            entities.append(
+                XiaomiAirPurifierFilterRemaining(
+                    NAME_FILTER_LIFE_REMAINING,
+                    air_purifier,
+                    config_entry,
+                    unique_id + "_filter_life",
+                )
+            )
+            entities.append(
+                XiaomiAirPurifierFilterUsed(
+                    NAME_FILTER_HOURS_USED,
+                    air_purifier,
+                    config_entry,
+                    unique_id + "_filter_used",
+                )
+            )
+        elif model in MODELS_PURIFIER_MIOT:
+            air_purifier = AirPurifierMiot(host, token)
+            entities.append(
+                XiaomiAirPurifier(
+                    NAME_AIR_QUALITY_INDEX, air_purifier, config_entry, unique_id
+                )
+            )
+            entities.append(
+                XiaomiAirPurifierFilterRemaining(
+                    NAME_FILTER_LIFE_REMAINING,
+                    air_purifier,
+                    config_entry,
+                    unique_id + "_filter_life",
+                )
+            )
+            entities.append(
+                XiaomiAirPurifierFilterUsed(
+                    NAME_FILTER_HOURS_USED,
+                    air_purifier,
+                    config_entry,
+                    unique_id + "_filter_used",
+                )
+            )
+        elif model.startswith("zhimi.airpurifier."):
+            air_purifier = AirPurifier(host, token)
+            entities.append(
+                XiaomiAirPurifier(
+                    NAME_AIR_QUALITY_INDEX, air_purifier, config_entry, unique_id
+                )
+            )
+            entities.append(
+                XiaomiAirPurifierFilterRemaining(
+                    NAME_FILTER_LIFE_REMAINING,
+                    air_purifier,
+                    config_entry,
+                    unique_id + "_filter_life",
+                )
+            )
+            entities.append(
+                XiaomiAirPurifierFilterUsed(
+                    NAME_FILTER_HOURS_USED,
+                    air_purifier,
+                    config_entry,
+                    unique_id + "_filter_used",
+                )
+            )
+        else:
+            device = AirQualityMonitor(host, token)
+            entities.append(
+                XiaomiAirQualityMonitor(name, device, config_entry, unique_id)
+            )
 
     async_add_entities(entities, update_before_add=True)
 
 
+class XiaomiAirPurifierFilterUsed(XiaomiMiioEntity, SensorEntity):
```

Comment:
I would try to re-write this so that you only have 1 sensor class, use input arguments to define which attribute to use etc.
There are way to many attributes for Miio defices to have a seperate class for each attribute, I think we would end up with 25+ classes which is just not maintanable in the long run.
Besides there is a lot of code duplication between these new classes.